lib/lighthouse/benefits_documents/form_526_lighthouse_documents_service.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'logging/third_party_transaction'
+require 'lighthouse/benefits_documents/worker_service'
+
+module BenefitsDocuments
+  module Form526LighthouseDocumentsService
+    extend Logging::ThirdPartyTransaction::MethodWrapper
+
+    wrap_with_logging(:upload_from_client)
```

Comment:
@SamStuckey quick questions here as I get into the DataDog dashboards.

 I'm thinking I should be monitoring the response code and body returned from this `upload_from_client` method, along with the `document_type` argument passed to `upload_lighthouse_document`. Assume there is a way to create a widget for each of the document types I'm using this method for:
- Supplemental documents (the ones the vet uploads that are handled in the SubmitUploads worker)
- Form 0781
- Form 0781a
- BDD Instructions

So initial questions:
1. Do you have an example of a dashboard you've created that leverages your 3PI logging behavior?
2. For setting up dashboards for all of those use cases (recall this is a module I'm reusing in a few places), would it make more sense to update the code here to this:

```ruby
  wrap_with_logging(
      :upload_from_client
      additional_instance_logs: {
        document_type: [:document_type]
      }
    )
```

where `document_type` becomes a new instance var, and set up the widget that way.

OR

Is there a way in DataDog to set up a widget that tracks this by caller? Eg. `EVSS::DisabilityCompensationForm::SubmitForm0781#upload_lighthouse_document` would be one.

LMK if this would be easier to chat about in a call, but basically wondering if I should make changes here for the purposes of my monitoring. I'm guessing the second option is easier and makes more sense.

Reply:
(also for reference if you follow the below code all the way through the Lighthouse API wrapper I'm using the mobile team built, this returns a `Faraday::Response` object)
```
    def upload_from_client(user_icn, file_body, lighthouse_document)
      client = BenefitsDocuments::WorkerService.new(user_icn)
      client.upload_document(file_body, lighthouse_document)
    end
```

Reply:
@SamStuckey quick follow up here, TLDR starting to plan out my monitoring dashboards for this release and wondering if there are any additional changes to the monitoring here that may make that easier. Still learning how team 1 has been setting up these dashboards but wondering what your experience has been/maybe we set one up together if that makes more sense thanks!